pkg/util/kubernetes/apiserver/leaderelection/leaderelection.go

```diff
@@ -142,16 +145,21 @@ func (le *LeaderEngine) init() error {
 	}
 
 	le.coreClient = apiClient.Cl.CoreV1()
-	// Will be required once we migrate to Kubernetes deps >= 0.24
-	le.coordClient = nil
+	le.coordClient = apiClient.Cl.CoordinationV1()
 
-	// check if we can get ConfigMap.
-	_, err = le.coreClient.ConfigMaps(le.LeaderNamespace).Get(context.TODO(), le.LeaseName, metav1.GetOptions{})
-	if err != nil && errors.IsNotFound(err) == false {
-		log.Errorf("Cannot retrieve ConfigMap from the %s namespace: %s", le.LeaderNamespace, err)
+	usingLease, err := CanUseLease(apiClient.DiscoveryCl)
```

Comment:
Since the addition of missing leases RBAC in the Helm chart are quite new, and the fact that the datadog-operator supporting it is not yet released.  I think it is safer to keep `configmap` the default object use for leader election. 

My idea is to add a new parameter in the config package (near https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L715-L717) to default the object to `configmap` 

```golang
config.BindEnvAndSetDefault("leader_election_default_resource","configmap")
```

We keep this option defaulted to configmap for a while. and then we can empty the default value. 
The behaviour that I think make sense is 

* `if leader_election_default_resource == "configmap"` use `configmap` without checking if "lease" object is present
* `if leader_election_default_resource == "lease"`  use "lease" without checking if "lease" object is present
* `if leader_election_default_resource == ""` check the present of "lease" resource, if present use it, else use configmap.


By doing this it will allow us to be backward compatible, but also allow to default to "" in the helm chart to use the new logique.

Reply:
It's now done. I'll open a PR to update our helm chart